app/src/main/java/com/sdp/movemeet/utility/DistanceCalculator.java

```diff
@@ -47,18 +61,28 @@ public void setActivities(ArrayList<Activity> activities) {
         }
     }
 
-
+    /**
+     * Called in MainMapFragment to return the list of N closest activities.
+     * @param n the number of activities to return
+     */
     @RequiresApi(api = Build.VERSION_CODES.N)
     public ArrayList<Activity> getTopActivities(Integer n) {
         if (sorted) {
-            Stream<Activity> topActivitiesStream = activityDistanceMap.subList(0, n).stream().map(Pair::getKey);
-            return topActivitiesStream.collect(Collectors.toCollection(ArrayList::new));
-
+            if (n >= activityDistanceMap.size()) {
+                return getAllActivities();
```

Comment:
You can directly use getActivities() from ActivitiesUpdater that returns the list of all activities. Alternatively, you could store the list of activities obtained from setActivities, and return it. No need to compute from the list of pairs in a for loop, as you will return the whole list anyway.

Reply:
That's a good point! I will refactor it, because it can help us performance-wise (the map will load quicker).